src/normalizeBenchmarkResult.ts

```diff
@@ -0,0 +1,30 @@
+import { BenchmarkResult } from './extract';
+import { normalizeUnit } from './normalizeUnit';
+import { extractRangeInfo } from './extractRangeInfo';
+
+export function normalizeBenchmarkResult(
+    prevBenchResult: BenchmarkResult | undefined | null,
+    currenBenchResult: BenchmarkResult,
+): BenchmarkResult {
+    if (!prevBenchResult) {
+        return currenBenchResult;
+    }
+
+    const prevUnit = prevBenchResult.unit;
+    const currentUnit = currenBenchResult.unit;
+    const currentRange = currenBenchResult.range;
+    const currentRangeInfo = extractRangeInfo(currentRange);
+
+    const normalizedValue = normalizeUnit(prevUnit, currentUnit, currenBenchResult.value);
```

Comment:
suggestion: the naming here could be better, you are normalizing the value by calling a function named `normalizeUnit` and one line below you are creating a `normalizedUnit` variable.

Maybe a better name for the function is `normalizeValueByUnit` or something similar.